### PR TITLE
Clean up server discovery code

### PR DESCRIPTION
**Pull request by Bond-009**
* Use the raw socket directly instead of custom abstractions
* Use newer `async` api

### CODE REVIEW

File: Emby.Server.Implementations/Udp/UdpServer.cs [hunk 1]
```diff
@@ -1,112 +1,47 @@
...
-        private bool _isDisposed;
-
-        private readonly List<Tuple<string, bool, Func<string, IPEndPoint, Encoding, CancellationToken, Task>>> _responders = new List<Tuple<string, bool, Func<string, IPEndPoint, Encoding, CancellationToken, Task>>>();
-
         private readonly IServerApplicationHost _appHost;
-        private readonly IJsonSerializer _json;
 
         /// <summary>
-        /// Initializes a new instance of the <see cref="UdpServer" /> class.
+        /// The _udp client.
```
Comment (cvium) [inline Emby.Server.Implementations/Udp/UdpServer.cs]: Is this necessary? And if you comment one of the fields, you should comment them all 😄

Comment (dkanada) [reply Emby.Server.Implementations/Udp/UdpServer.cs]: Do you mean this specific comment, or comments in general? This is actually why I am wary of requiring documentation. It alsmost always leads to comments like these.

Comment (Bond-009) [reply Emby.Server.Implementations/Udp/UdpServer.cs]: This is an old comment, I didn’t write it

Comment (cvium) [reply Emby.Server.Implementations/Udp/UdpServer.cs]: The diff didn't show that

**Pull request merged**